R/pucAggregate.R

```diff
@@ -20,14 +22,28 @@
 #' }
 #' @importFrom withr with_tempdir local_package
 #' @importFrom utils untar modifyList
+#' @importFrom callr r
+#' @importFrom renv activate restore
 #' @export
-pucAggregate <- function(puc, regionmapping = getConfig("regionmapping"), ...) {
+pucAggregate <- function(puc, regionmapping = getConfig("regionmapping"), ..., renv = TRUE) {
   argumentValues <- c(as.list(environment()), list(...)) # capture arguments for logging
   extraArgs <- list(...)
-  startinfo <- toolstartmessage("pucAggregate", argumentValues, 0)
+  startinfo <- toolstartmessage("pucAggregate", argumentValues, "+")
   puc <- normalizePath(puc)
   if (file.exists(regionmapping)) regionmapping <- normalizePath(regionmapping)
 
+  .aggregatePuc <- function(regionmapping, cfg, madratCfg, nestinglevel) {
+    if (file.exists("puc/renv.lock")) {
+      renv::activate()
+      renv::restore(lockfile = "puc/renv.lock", prompt = FALSE)
+    }
+    withr::local_options(madrat_cfg = madratCfg, gdt_nestinglevel = nestinglevel)
+    if (!is.null(cfg$package)) withr::local_package(cfg$package)
+    madrat::setConfig(regionmapping = regionmapping, forcecache = TRUE,
+                      .verbose = FALSE, .local = TRUE)
+    do.call(retrieveData, c(cfg$args, list(renv = FALSE)))
```

Comment:
I'm surprised this is working without `madrat::` prefix?
```suggestion
    do.call(madrat::retrieveData, c(cfg$args, list(renv = FALSE)))
```